src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticateResult.cs

```diff
@@ -31,12 +31,17 @@ public class AuthenticateResult
         /// <summary>
         /// Additional state values for the authentication session.
         /// </summary>
-        public AuthenticationProperties Properties => Ticket?.Properties;
+        public AuthenticationProperties Properties => Ticket?.Properties ?? Error?.Properties;
```

Comment:
Then maybe instead of an Error property, we should have ErrorResult derive from AuthenticateResult, have Ticket return null, Properties return the error properties.